src/contract-deployment/config.ts

```diff
@@ -43,7 +43,17 @@ export const makeContractDeployConfig = async (
   return {
     OVM_L1CrossDomainMessenger: {
       factory: getContractFactory('OVM_L1CrossDomainMessenger'),
-      params: [AddressManager.address],
+      params: [],
+    },
+    Proxy__OVM_L1CrossDomainMessenger: {
+      factory: getContractFactory('Lib_ResolvedDelegateProxy'),
+      params: [AddressManager.address, 'OVM_L1CrossDomainMessenger'],
+      afterDeploy: async (contracts): Promise<void> => {
+        const xDomainMessenger = getContractFactory(
+          'OVM_L1CrossDomainMessenger'
+        ).attach(contracts.Proxy__Proxy__OVM_L1CrossDomainMessenger.address)
```

Comment:
Why is it `Proxy__Proxy`?

Reply:
😭 Lemme add an issue for removing the `Proxy__` prefix. Right now all contracts are incorrectly prefixed with `Proxy__` so we need to remove that prefix, but think that should be in another PR

Reply:
https://github.com/ethereum-optimism/contracts-v2/issues/55 -- here we go!